x-pack/plugins/apm/public/components/app/Settings/agent_keys/create_agent_key.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButton,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiFieldText,
+  EuiText,
+  EuiFormFieldset,
+  EuiCheckbox,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash';
+import { callApmApi } from '../../../../services/rest/createCallApmApi';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import { ApmPluginStartDeps } from '../../../../plugin';
+import { CreateApiKeyResponse } from '../../../../../common/agent_key_types';
+
+interface Props {
+  onCancel(): void;
+  onSuccess: (agentKey: CreateApiKeyResponse) => void;
+  onError: (keyName: string) => void;
+}
+
+export function CreateAgentKeyFlyout({ onCancel, onSuccess, onError }: Props) {
+  const {
+    services: { security },
+  } = useKibana<ApmPluginStartDeps>();
+
+  const [username, setUsername] = useState('');
+
+  const [formTouched, setFormTouched] = useState(false);
+  const [keyName, setKeyName] = useState('');
+  const [agentConfigChecked, setAgentConfigChecked] = useState(true);
+  const [eventWriteChecked, setEventWriteChecked] = useState(true);
+  const [sourcemapChecked, setSourcemapChecked] = useState(true);
```

Comment:
WDYT about combining these in a single state?

```suggestion
  const [agentKeyBody, setAgentKeyBody] = useState<AgentKeyBody>({
    name: '',
    sourcemap: true,
    event: true,
    agentConfig: true
  });
```

Reply:
It's a matter of preference and code consistency when working with forms, I normally use different states but happy to change it if a single state is preferred

Reply:
Changed to single state in https://github.com/elastic/kibana/pull/120765